nodebuilder/blobstream/service.go

```diff
@@ -0,0 +1,534 @@
+// Note: queries the whole EDS to generate the proof.
+// This can be improved by selecting the set of shares that will need to be used to create
+// the proof and only querying them. However, that would require re-implementing the logic
+// in Core. Also, core also queries the whole EDS to generate the proof. So, it's fine for
+// now. In the future, when blocks get way bigger, we should revisit this and improve it.
+func (s *Service) ProveShares(ctx context.Context, height uint64, start, end uint64) (*ResultShareProof, error) {
+	log.Debugw("proving share range", "start", start, "end", end, "height", height)
+	if height == 0 {
+		return nil, fmt.Errorf("height cannot be equal to 0")
+	}
+	if start == end {
+		return nil, fmt.Errorf("start share cannot be equal to end share")
+	}
+	if start > end {
+		return nil, fmt.Errorf("start share %d cannot be greater than end share %d", start, end)
+	}
+
+	log.Debugw("getting extended header", "height", height)
+	extendedHeader, err := s.headerServ.GetByHeight(ctx, height)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugw("getting eds", "height", height)
+	eds, err := s.shareServ.GetEDS(ctx, extendedHeader)
+	if err != nil {
+		return nil, err
+	}
+
+	startInt, err := uint64ToInt(start)
+	if err != nil {
+		return nil, err
+	}
+	endInt, err := uint64ToInt(end)
+	if err != nil {
+		return nil, err
+	}
+	odsShares, err := shares.FromBytes(eds.FlattenedODS())
+	if err != nil {
+		return nil, err
+	}
+	nID, err := pkgproof.ParseNamespace(odsShares, startInt, endInt)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugw("generating the share proof", "start", start, "end", end, "height", height)
+	proof, err := pkgproof.NewShareInclusionProofFromEDS(eds, nID, shares.NewRange(startInt, endInt))
+	if err != nil {
+		return nil, err
+	}
+	return &ResultShareProof{ShareProof: proof}, nil
+}
+
+// ProveCommitment generates a commitment proof for a share commitment.
+// It takes as argument the height of the block containing the blob of data, its
+// namespace and its share commitment.
+func (s *Service) ProveCommitment(ctx context.Context, height uint64, namespace share.Namespace, shareCommitment []byte) (*ResultCommitmentProof, error) {
+	log.Debugw("proving share commitment", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	if height == 0 {
+		return nil, fmt.Errorf("height cannot be equal to 0")
+	}
+
+	// get the blob to compute the subtree roots
+	log.Debugw("getting the blob", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	blb, err := s.blobServ.Get(ctx, height, namespace, shareCommitment)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugw("converting the blob to shares", "height", height, "commitment", shareCommitment, "namespace", namespace)
+	blobShares, err := blob.BlobsToShares(blb)
+	if err != nil {
+		return nil, err
+	}
+	if len(blobShares) == 0 {
+		// TODO we return the share commitment as hex or some other format?
```

Comment:
[note to reviewers]
should we return the share commitment as hex or some other format?

Reply:
```suggestion
```